pkg/kubelet/kubelet.go

```diff
@@ -913,7 +923,7 @@ func (kl *Kubelet) GenerateRunContainerOptions(pod *api.Pod, container *api.Cont
 			opts.PodContainerDir = p
 		}
 	}
-	if pod.Spec.DNSPolicy == api.DNSClusterFirst {
+	if pod.Spec.DNSPolicy == api.DNSClusterFirst || kl.resolverConfig != ResolvConfDefault {
```

Comment:
I don't understand this conditional.  Even if kl.resolverConfig == ResolvConfDefault, we still want to do this.  The only exception would be if kl.resolverConfig == "", right?